js/ui/number_box/number_box.mask.js

```diff
@@ -56,19 +57,32 @@ var NumberBoxMask = NumberBoxBase.inherit({
             leftArrow: that._arrowHandler.bind(that, MOVE_FORWARD),
             rightArrow: that._arrowHandler.bind(that, MOVE_BACKWARD),
             home: that._moveCaretToBoundary.bind(that, MOVE_FORWARD),
-            enter: that._setTextByParsedValue.bind(that),
+            enter: that._updateFormattedValue.bind(that),
             end: that._moveCaretToBoundary.bind(that, MOVE_BACKWARD)
         });
     },
 
     _focusOutHandler: function(e) {
         if(this._useMaskBehavior()) {
-            this._setTextByParsedValue();
+            this._updateFormattedValue();
         }
 
         this.callBase(e);
     },
 
+    _updateFormattedValue: function() {
+        this._setTextByParsedValue();
+
+        if(this._isValueDirty()) {
+            eventsEngine.trigger(this._input(), "change");
+        }
+    },
+
+    _isValueDirty: function() {
+        //https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/15181565/
+        return browser.msie && this._parsedValue !== this._formattedValue;
```

Comment:
parsedValue will not be equal to firmatted value if stubs are used in format. For example: `$ #0`. ParsedValue will 123 but formatted will $ 123